how-to-contribute.md

```diff
@@ -1,191 +1,147 @@
+-----
 
-The Flink project accepts code contributions though the [GitHub Mirror](https://github.com/apache/flink), in the form of [Pull Requests](https://help.github.com/articles/using-pull-requests). Pull requests are basically a simpler way of offering a patch, by providing a pointer to a code branch that contains the change.
+## Propose an improvement or a new feature
 
-It is also possible to attach a patch to a [JIRA]({{site.FLINK_ISSUES_URL}}) issue.
+Our community is constantly looking for feedback to improve Apache Flink. If you have an idea how to improve Flink or have a new feature in mind that would be beneficial for Flink users, please open an issue in [Flink's JIRA](http://issues.apache.org/jira/browse/FLINK). The improvement or new feature should be described in appropriate detail and include the scope and its requirements if possible. Detailed information is important for a few reasons:
 
+- It ensures your requirements are met when the improvement or feature is implemented.
+- It helps to estimate the effort and to design a solution that addresses your needs.
+- It allow for constructive discussions that might arise around this issue.
 
-### Setting up the Infrastructure and Creating a Pull Request
+Detailed information is also required, if you plan to contribute the improvement or feature you proposed yourself. Please read the [Contribute code]({{ site.base }}/contribute-code.html) guide in this case as well. 
 
-1. The first step is to create yourself a copy of the Flink code base. We suggest to fork the [Flink GitHub Mirror Repository](https://github.com/apache/flink) into your own [GitHub](https://github.com) account. You need to register on GitHub for that, if you have no account so far.
+-----
 
-2. Next, clone your repository fork to your local machine.
-  
-   ```
-git clone https://github.com/<your-user-name>/flink.git
-```
+## Help others and join the discussions
 
-3. It is typically helpful to switch to a *topic branch* for the changes. To create a dedicated branch based on the current master, use the following command:
+Most communication in the Apache Flink community happens on two mailing lists:
 
-    ```
-git checkout -b myBranch master
-```
+- The user mailing lists `user@flink.apache.org` is the place where users of Apache Flink ask questions and seek for help or advice. Joining the user list and helping other users is a very good way to contribute to Flink's community. Furthermore, there is the [[Flink]](http://stackoverflow.com/questions/tagged/flink) tag on Stack Overflow if you'd like to help Flink users (and harvest some points) there.
```

Comment:
The SO tags got merged with "apche-flink" being the main tag -> "https://stackoverflow.com/questions/tagged/apache-flink"